packages/pods-core/src/builtin/AirtablePod.ts

```diff
@@ -34,40 +36,36 @@ export class AirtableExportPod extends ExportPod<AirtableExportConfig> {
   static id: string = ID;
   static description: string = "export notes to airtable";
 
-  get config() {
-    return super.config.concat([
-      {
-        key: "tableName",
-        description: "Name of the airtable",
-        type: "string",
-        required: true,
+  get config(): JSONSchemaType<AirtableExportConfig> {
+    return PodUtils.createExportConfig({
+      required: [
+        "tableName",
+        "srcHierarchy",
+        "baseId",
+        "apiKey",
+        "srcFieldMapping",
+      ],
+      properties: {
+        tableName: { type: "string", description: "Name of the airtable" },
+        srcHierarchy: {
+          type: "string",
+          description: "The src .md file from where to start the sync",
+        },
+        apiKey: {
+          type: "string",
+          description: "Api key for airtable",
+        },
+        baseId: {
+          type: "string",
+          description: " base Id of airtable base",
+        },
+        srcFieldMapping: {
+          type: "object",
+          description:
+            "mapping of airtable fields with the note eg: {Created On: created, Notes: body}",
+        },
       },
-      {
-        key: "srcHierarchy",
-        description: "The src .md file from where to start the sync",
-        type: "string",
-        required: true,
-      },
-      {
-        key: "apiKey",
-        description: "Api key for airtable",
-        type: "string",
-        required: true,
-      },
-      {
-        key: "baseId",
-        description: " base Id of airtable base",
-        type: "string",
-        required: true,
-      },
-      {
-        key: "srcFieldMapping",
-        description:
-          "mapping of airtable fields with the note eg: {Created On: created, Notes: body}",
-        type: "object",
-        required: true,
-      },
-    ]);
+    }) as JSONSchemaType<AirtableExportConfig>;
```

Comment:
instead of doing a cast, can `createExportPodConfig` take a generic `T extends ExportPodConfig`?

Reply:
Actually no, this results in two errors : incompatible "type" error and value of T never used.
Also, to resolve the " value of T never used" error, if we keep the return type of  createExportPodConfig to be : JSONSchemaType<T>, ajv doesn't allow a generic schema to have a type as "object".

```js
static createExportConfig<T extends ExportPodConfig>(opts: { required: string[]; properties: any }) : JSONSchemaType<T>
 {
    return {
      type: "object",
      ...
}
}
```